vars/treebuild.groovy

```diff
@@ -26,11 +26,104 @@ def upstreamBuild(List<String> projectCollection, String currentProject, String
+    }
+}
+
+/**
+ * Fetches the goals from a Jenkins properties file
+ * @param project - current project
+ * @param propertiesFileId - pointing to a Jenkins properties file
+ * @param type (can be "current" or "upstream")
+ */
+def getGoals(String project, String propertiesFileId, String type = 'current') {
+    configFileProvider([configFile(fileId: propertiesFileId, variable: 'PROPERTIES_FILE')]) {
+        def propertiesFile = readProperties file: PROPERTIES_FILE
+        return propertiesFile."goals.${project}.${type}" ?: propertiesFile."goals.default.${type}"
+    }
+}
+
+/**
+ * Fetches the artifacts from a Jenkins properties file
+ * @param project - current project
+ * @param propertiesFileId - pointing to a Jenkins properties file
+ */
+def getArtifacts(String project, String propertiesFileId) {
+    configFileProvider([configFile(fileId: propertiesFileId, variable: 'PROPERTIES_FILE')]) {
+        def propertiesFile = readProperties file: PROPERTIES_FILE
+        return propertiesFile."artifacts.${project}" ?: propertiesFile."artifacts.default"
+    }
+}
+
+/**
+ * Builds the compile downstream (upstream projects, the current project and the downstream projects)
+ * @param the file with a collection of items following the pattern PROJECT_GROUP/PROJECT_NAME, for example kiegroup/drools
+ * @param currentProject the project to build the stream from, like kiegroup/drools
+ * @param settingsXmlId maven settings xml file id
+ * @param propertiesFileId file that defines the maven goals for each rep
+ */
+def pullCompileDownstreamBuild(List<String> projectCollection, String currentProject, String settingsXmlId, String propertiesFileId) {
+    println "Compile downstream build of project [${currentProject}] for project collection ${projectCollection}"
+    checkoutProjects(projectCollection)
+
+    def currentProjectIndex = projectCollection.findIndexOf { it == currentProject }
+    // Build project tree from currentProject node
+    for (i = 0; i < projectCollection.size(); i++) {
+        def type = i < currentProjectIndex ? 'upstream' :  i == currentProjectIndex ? 'current' : 'downstream'
+        println "Build of project [${projectCollection.get(i)}] with type [${type}]"
+        buildProject(projectCollection.get(i), settingsXmlId, getGoals(projectCollection.get(i), propertiesFileId, type))
+    }
+}
+
+/**
+ *
+ * @param project a string following the pattern PROJECT_GROUP/PROJECT_NAME, for example kiegroup/drools
+ * @param settingsXmlId maven settings xml file id
+ * @param goals maven goals
+ * @param sonarCloudId token for sonarcloud
+ */
+def buildProjectSonar(String project, String settingsXmlId, String goals, String sonarCloudId) {
+    def projectGroupName = getProjectGroupName(project)
+    def group = projectGroupName[0]
+    def name = projectGroupName[1]
+
+    println "Building ${group}/${name}"
+    dir("${env.WORKSPACE}/${group}_${name}") {
+        maven.runMavenWithSettingsSonar(settingsXmlId, goals, sonarCloudId)
```

Comment:
In case you change the `runMavenWithSettingsSonar` method you just change this by `maven.runMavenWithSettingsSonar(settingsXmlId, goals, sonarCloudId, "${group}_${name}.maven.log")`

Reply:
done